tests/http/federation/test_matrix_federation_agent.py

```diff
@@ -100,24 +85,39 @@ def setUp(self):
             had_well_known_cache=self.had_well_known_cache,
         )
 
-        self.agent = MatrixFederationAgent(
-            reactor=self.reactor,
-            tls_client_options_factory=self.tls_factory,
-            user_agent="test-agent",  # Note that this is unused since _well_known_resolver is provided.
-            ip_blacklist=IPSet(),
-            _srv_resolver=self.mock_resolver,
-            _well_known_resolver=self.well_known_resolver,
-        )
+        self.agent = self._make_agent()
```

Comment:
I think it would be better not to create an agent here at all, than to create it and then recreate for some of the tests.